src/utils/tooltipify.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from "react";
+import ReactDOM from 'react-dom';
+
+import PlatformPeg from "../PlatformPeg";
+import TextWithTooltip from "../components/views/elements/TextWithTooltip";
+
+/**
+ * Recurses depth-first through a DOM tree, adding tooltip previews for link elements.
+ *
+ * @param {Element[]} rootNodes - a list of sibling DOM nodes to traverse to try
+ *   to add tooltips.
+ * @param {Element[]} ignoredNodes: a list of nodes to not recurse into.
+ * @param {Element[]} containers: an accumulator of the DOM nodes which contain
+ *   React components that have been mounted by this function. The initial caller
+ *   should pass in an empty array to seed the accumulator.
+ */
+export function tooltipifyLinks(rootNodes: ArrayLike<Element>, ignoredNodes: Element[], containers: Element[]) {
```

Comment:
Any chance we can get tests for this?

Reply:
I've added a few simplistic snapshot tests but not sure if these are great. Happy to take feedback.

Reply:
I'm having a similar issue as on https://github.com/matrix-org/matrix-js-sdk/pull/2272 now where Sonarcloud [reports](https://sonarcloud.io/component_measures?id=matrix-react-sdk&metric=new_lines&pullRequest=8394&view=list) 200k new lines on my PR after a merge. All I did was:

```
git checkout develop
git pull
git checkout feature/url-tooltip
git merge develop
git push
```

Am I doing something wrong here?

Reply:
Update your fork's develop branch too, as that's what it is currently (wrongly) comparing to

Reply:
Ah geez, now it all makes sense. 🤦‍♂️